pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -489,6 +493,32 @@ func (r *Reconciler) getOrCreateClusterResourceSnapshot(ctx context.Context, crp
 	return latestResourceSnapshot, nil
 }
 
+// splitSelectedResources splits selected resources in a ClusterResourcePlacement into separate lists
+// so that the total size of each split list of selected Resources is within 1MB limit.
+func splitSelectedResources(selectedResources []fleetv1beta1.ResourceContent) ([][]fleetv1beta1.ResourceContent, error) {
+	var selectedResourcesList [][]fleetv1beta1.ResourceContent
+	i := 0
+	for i < len(selectedResources) {
+		j := i
+		currentSize := 0
+		var snapshotResources []fleetv1beta1.ResourceContent
+		for j < len(selectedResources) {
+			currentSize += len(selectedResources[j].Raw)
+			if currentSize > resourceSnapshotResourceSizeLimit {
+				break
+			}
+			snapshotResources = append(snapshotResources, selectedResources[j])
+			j++
+		}
+		if len(snapshotResources) == 0 {
```

Comment:
This assumption is not correct.

Our limit is 1mb, but it's possible that a valid resource is more than 1mb as the etcd size limit is 1.5mb.

We should handle this valid case.